Site/dataobjects/SiteApiCredential.php

```diff
@@ -74,11 +74,38 @@ public function loadByApiKey($key, SiteInstance $instance =  null)
 			return false;
 
 		$this->initFromRow($row);
+
 		$this->generatePropertyHashes();
 
 		return true;
 	}
 
+	// }}}
+	// {{{ public function getPromotionByType()
+
+	public function getPromotionByType($type)
+	{
```

Comment:
Calling `$this->checkDB()` at the top of the method will ensure the record is connected to the database before doing the query.